counsel.el

```diff
@@ -4090,15 +4100,21 @@ candidate."
   (org-map-entries
    (lambda ()
      (let* ((components (org-heading-components))
-            (level (make-string
-                    (if org-odd-levels-only
-                        (nth 1 components)
-                      (nth 0 components))
-                    ?*))
-            (todo (nth 2 components))
-            (priority (nth 3 components))
+            (level (when (eq counsel-org-goto-display-style 'headline)
```

Comment:
Since we care about the expression's value, it's more correct to use `and` in place of `when` here and all corresponding cases below.